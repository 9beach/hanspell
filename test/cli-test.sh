#!/usr/bin/env bash

set -e

cat README.md | lib/cli.js > /dev/null 2>&1
cat test/cli-test.in | lib/cli.js -d 2>&1 | diff - test/cli-test.d.out
cat test/cli-test.in | lib/cli.js -p 2>&1 | diff - test/cli-test.p.out
cat test/cli-test.in | lib/cli.js -j 2>&1 | diff - test/cli-test.j.out
cat test/cli-test.in | lib/cli.js -a 2>&1 | diff - test/cli-test.a.out
