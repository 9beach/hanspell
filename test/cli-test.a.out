-- 한스펠 로그 시작
-- 교정 제안: 있나요 묻는다면 -> 있나요. 묻는다면
강조하는 말이나 지정하는 말에 쓰거나 대화나 직접 인용 또는 간접 인용에 쓰는 문장 부호인 큰따옴표("")나 작은따옴표('')는 뒤에 오는 조사나 어미와 붙여 씁니다.

  (예) 그가 '내일 만나자.'라고 말했다.
      그는 "나는 대통령이다."라고 말했다. '배부른 돼지'보다는 '배고픈 소크라테스'가 되겠다.
-- 교정 제안: 곤색이다 -> 감색이다
'곤색'은 '감색'으로 순화하여 씁니다.
-- 교정 제안: 곤색의 -> 감색의
'곤색'은 '감색'으로 순화하여 씁니다.
-- 교정 제안: 열두살 -> 열두 살
띄어쓰기 오류입니다. 대치어를 참고하여 고쳐 쓰세요.
-- 교정 제안: 양파자루 -> 양파 자루
띄어쓰기 오류입니다. 맞춤법 표준안에 따르면 명사와 명사는 복합명사를 구성할 수 있습니다. 그러나 의미상으로 옳지 않거나 사용 빈도가 아주 낮은 복합 명사는 띄어 쓰시면 좋습니다. 굳이 붙여 쓰고자 할 때는 붙여 써도 무방합니다.

  (예) 호랑이가수 (X) -> 호랑이 가수 (O)
      어머니기지 (X) -> 어머니 기지 (O)
      생산거지 (X)   -> 생산 거지 (O)
-- 교정 제안: (스완네 집) 에 -> (스완네 집)에
간혹 입력 잘못이나 교정의 잘못으로 붙여 써야 하는 낱말이나 어절 등이 띄어져 있을 때가 있습니다. 철자검사기가 이 낱말이나 어절을 분석할 수 없어 좌우 문맥 등을 참고하여 고쳐 본 것입니다. 문맥에 알맞게 붙이거나 띄어 쓰세요.
-- 교정 제안: 리랜드는 -> 이랜드는 (시랜드는)
철자 검사를 해 보니 이 어절은 분석할 수 없으므로 틀린 말로 판단하였습니다.

후보 어절은 이 철자검사/교정기에서 띄어쓰기, 붙여 쓰기, 음절대치와 같은 교정방법에 따라 수정한 결과입니다.

후보 어절 중 선택하시거나 오류 어절을 수정하여 주십시오.

* 단, 사전에 없는 단어이거나 사용자가 올바르다고 판단한 어절에 대해서는 통과하세요!!
-- 교정 제안: 한바퀴 -> 한 바퀴
수사 또는 수관형사와 단위성의존명사는 띄어 써야 합니다.

(예)
강아지가 새끼를 다섯 마리나 낳았다.
사람이 열 명 정도 모였다.
우리 마당에는 감나무 세 그루가 서 있다.
-- 교정 제안: 돌껀데 -> 돌 건데
'꺼'는 잘못된 표기로 의존명사 '것' 또는 '거'로 고쳐 쓰는 것이 옳습니다.

(예)
갖고 싶은 거 다 사도 돼요?
그 일은 내가 한 것이 아니다.
-- 교정 제안: 말했더만 -> 말하였더니만
'-더만'은 잘못된 어미입니다. '-더니만'을 써서 '말하였더니만'으로 고쳐 쓰세요.

(예)
어머니를 오시라고 했더니만 아버지 혼자 오셨군요.
그는 가던 길을 우뚝 멈추더니만 갑자기 뒤를 돌아보았다.
밤새 잠을 못 잤더니만 정신이 하나도 없네.
-- 교정 제안: 집) 에 -> 집)에
《한글 맞춤법》 제41항에 따라 조사는 그 앞말에 붙여 씁니다.
-- 한스펠 로그 끝
아마도 아프리카

이제니

코끼리 사자 기린 얼룩말 호랑이
멀리 있는 것들의 이름을 마음속으로 부를 때
나는 슬픈가 나는 위안이 필요한가
아마도 아프리카 아마도 아주 조금

호랑이, 그것은 나만의 것
따뜻하고 보드랍고 발톱이 없는 것

살고 있나요. 묻는다면 아마도 아프리카
아마도 나는 아주 조금 살고 있어요

내 머릿속은
반은 쑥색이고 반은 감색이다
쑥색과 감색의 접합점은 성홍열 같은 선홍색

열두 살 이후로 농담이 입에 배었다
옷에도 머리카락에도 손톱 끝에도
주황색 양파 자루 속엔 어제의 열매들
양파가 익어가는 속도로 너는 울었지

눈을 감아도 선홍색이 보이면
다시 코끼리 사자 기린 얼룩말 호랑이
너무나 멀리 있지만 아마도 이미 아프리카
나는 하룻밤 사이에도 많은 곳을 돌아다닌다

---

(스완네 집)에 오세요.
이랜드는 얼굴 골격이 굵은 게, 어머니 쪽을 닮았다.
한 바퀴 돌 건데 말하였더니만
